frontend/client_sdk/src/CodaSDK.d.ts

```diff
@@ -0,0 +1,78 @@
+export declare type publicKey = string;
+export declare type privateKey = string;
+export declare type globalSlot = number;
+export declare type uint64 = string;
+export declare type key = {
+    readonly privateKey: privateKey;
+    readonly publicKey: publicKey;
+};
+export declare type signature = {
+    readonly field: string;
+    readonly scalar: string;
+};
+export declare type signed<signable> = {
+    readonly publicKey: publicKey;
+    readonly signature: signature;
+    readonly payload: signable;
+};
+export declare type stakeDelegation = {
+    readonly to: publicKey;
+    readonly from: publicKey;
+    readonly fee: uint64;
+    readonly nonce: number;
+    readonly memo?: string;
+    readonly validUntil: globalSlot;
+};
+export declare type payment = {
+    readonly to: publicKey;
+    readonly from: publicKey;
+    readonly fee: uint64;
+    readonly amount: uint64;
+    readonly nonce: number;
+    readonly memo?: string;
+    readonly validUntil: globalSlot;
+};
+/**
+  * Generates a public/private keypair
+  */
+export declare const genKeys: () => key;
+/**
+  * Signs an arbitrary message
+  *
+  * @param key - The keypair used to sign the message
+  * @param message - An arbitrary string message to be signed
+  * @returns A signed message
+  */
+export declare const signMessage: (message: string, key: key) => signed<string>;
+/**
+  * Verifies that a signature matches a message.
+  *
+  * @param signedMessage - A signed message
+  * @returns True if the `signedMessage` contains a valid signature matching
+  * the message and publicKey.
+  */
+export declare const verifyMessage: (signedMessage: signed<string>) => boolean;
+/**
+  * Signs a payment transaction using a private key.
+  *
+  * This type of transaction allows a user to transfer funds from one account
+  * to another over the network.
+  *
+  * @param payment - An object describing the payment
+  * @param key - The keypair used to sign the transaction
+  * @returns A signed payment transaction
+  */
+export declare const signPayment: (payment: payment, key: key) => signed<payment>;
```

Comment:
similarly to above, we may just want this to return the signature for convenience sake, so ppl won't have to unwrap the result

Reply:
If you only want the signature:
```js
let {signature} = signPayment(payment, keypair);
```
Not a super uncommon pattern, see react hooks etc.

But they'll probably want the payload too when submitting the txn to GraphQL.
```js
let sendMutation = (query, variables) => {
  fetch(
    "https://graphql.o1test.net/graphql", {
      method: "POST",
      body: JSON.stringify({ query, variables })
    }).then((result) => result.json());
  };

const query = gql`
  mutation SendSigned($payload: SendPaymentInput!, $signature: SignatureInput!) {
    sendPayment(input: $payload, signature: $signature) {
      payment { id }
    }
}`;

// create the urql request object
const request = createRequest(query, signPayment(payment, keypair));
```

Reply:
that's fine. I mean, they already have the payload, so all they need for the above request is the sig...